app/controllers/users_controller.rb

```diff
@@ -0,0 +1,9 @@
+class UsersController < ApplicationController
+  def index; end
+
+  def show; end
+
+  def posts; end
+
+  def show_post; end
```

Comment:
- [ ] This is not required for the users controller.

- [ ] Kindly make sure to make a file posts_controller and define it in the same way as users controller. Example:
```ruby
class PostsController < ApplicationController
  
  def index; end

  def show; end

end
```